subprojects/smoke-test/build.gradle.kts

```diff
@@ -93,40 +64,87 @@ tasks {
      */
     val santaGitUri = "https://github.com/gradle/santa-tracker-android.git"
 
-    register<RemoteProject>("santaTrackerKotlin") {
+    val santaTrackerKotlin by registering(RemoteProject::class) {
         remoteUri.set(santaGitUri)
         // Pinned from branch agp-3.6.0
         ref.set("3122343674246eaa56a5dd6365ea137edb021780")
     }
 
-    register<RemoteProject>("santaTrackerJava") {
+    val santaTrackerJava by registering(RemoteProject::class) {
         remoteUri.set(santaGitUri)
         // Pinned from branch agp-3.6.0-java
         ref.set("d8543e51ac5a4803a8ac57f0639229736f11e0a8")
     }
 
-    register<RemoteProject>("gradleBuildCurrent") {
+    val gradleBuildCurrent by registering(RemoteProject::class) {
         remoteUri.set(rootDir.absolutePath)
         ref.set(moduleIdentity.gradleBuildCommitId)
     }
 
-    val remoteProjects = withType<RemoteProject>()
+    val santaProjects = arrayOf(santaTrackerKotlin, santaTrackerJava)
+    val remoteProjects = santaProjects + gradleBuildCurrent
 
     if (BuildEnvironment.isCiServer) {
-        remoteProjects.configureEach {
-            outputs.upToDateWhen { false }
+        remoteProjects.forEach { remoteProject ->
+            remoteProject {
+                outputs.upToDateWhen { false }
+            }
         }
     }
 
     register<Delete>("cleanRemoteProjects") {
-        delete(remoteProjects.map { it.outputDirectory })
+        remoteProjects.forEach { remoteProject ->
+            delete(remoteProject.map { it.outputDirectory })
+        }
     }
 
-    withType<SmokeTest>().configureEach {
-        dependsOn(remoteProjects)
+    fun SmokeTest.configureForSmokeTest(vararg remoteProjects: TaskProvider<RemoteProject>) {
+        group = "Verification"
+        testClassesDirs = smokeTestSourceSet.output.classesDirs
+        classpath = smokeTestSourceSet.runtimeClasspath
+        maxParallelForks = 1 // those tests are pretty expensive, we shouldn"t execute them concurrently
         inputs.property("androidHomeIsSet", System.getenv("ANDROID_HOME") != null)
-        inputs.files(Callable { remoteProjects.map { it.outputDirectory } })
+        dependsOn(*remoteProjects)
+        inputs.files(Callable { remoteProjects.map { it.map { it.outputDirectory } } })
```

Comment:
Why do you need to use `Callable` here? remoteProjects.map { it.map { it.outputDirectory } }` is already lazy, since it is a list of Providers, right? Moreover, does this input declaration automatically carry the dependency information, so we don't need the `dependsOn(*remoteProjects)`?

Reply:
Right! I just confirmed it works and pushed. PTAL